### PR TITLE
Preliminary Meson support

### PR DESCRIPTION
**Pull request by deepbluev7**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -112,6 +112,30 @@ build linux arm64:
...
+  needs:
+    - job: "build synapse amd64"
+      optional: true
+  variables:
+    M_ARCH: x86_64
+    COVERAGE: "ON"
+  services: !reference [.build-linux, services]
+  before_script:
+    - echo 'https://dl-cdn.alpinelinux.org/alpine/edge/testing' >> /etc/apk/repositories
+    - apk update && apk add meson git g++ cmake pkgconf openssl openssl-dev make
```
Comment (redsky17) [inline .gitlab-ci.yml:128]: Do you need openssl and openssl-dev if you're using the meson wrap for them?

File: subprojects/packagefiles/openssl-meson/meson.build [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    '-Dincludedir_ignore=@PREFIX@/@INCLUDEDIR@',
+    target_machine.system() + '-' + target_machine.cpu_family(),
+                      ],
+  verbose: true,
+)
+
+ssl_dep = openssl.dependency('ssl')
+crypto_dep = openssl.dependency('crypto')
+openssl_dep = declare_dependency(dependencies: [ssl_dep, crypto_dep])
+meson.override_dependency('openssl', ssl_dep)
```
Comment (redsky17) [inline subprojects/packagefiles/openssl-meson/meson.build:21]: should this be `openssl_dep` instead of `ssl_dep`?

File: .gitlab-ci.yml (hunk 1 above)
Comment (deepbluev7) [reply .gitlab-ci.yml:128]: Yes, the wrap doesn't work for the cmake projects like curl. So basically it is only useful on platforms, where curl doesn't need openssl.

File: subprojects/packagefiles/openssl-meson/meson.build (hunk 2 above)
Comment (deepbluev7) [reply subprojects/packagefiles/openssl-meson/meson.build]: That file wasn't even used anymore, but yes, you would be right otherwise.

**Pull request merged**